app/controllers/incomeSources/add/IncomeSourcesAccountingMethodController.scala

```diff
@@ -116,7 +119,7 @@ class IncomeSourcesAccountingMethodController @Inject()(val authenticate: Authen
           routes.AddBusinessAddressController.show().url
       case UkProperty.key =>
           routes.AddIncomeSourceStartDateCheckController.show(UkProperty.key, isAgent, isChange = false).url
-      case ForeignProperty.key =>
+      case _ =>
```

Comment:
I would say this that we should have conversion from String => IncomeSourceType before pattern matching and raise failure if conversion is not succesfull